tools/testbench/testbench.c

```diff
@@ -269,17 +269,23 @@ int main(int argc, char **argv)
 		fprintf(stderr, "error: parsing topology\n");
 		exit(EXIT_FAILURE);
 	}
+	printf("Test Pipeline:\n");
+	printf("%s\n", pipeline);
```

Comment:
Check the debug message from parse_topology here.

According to the output, the pipeline graph is kind of weird... I'm sure if the message producer is correct...?
Test Pipeline:
PCM0P->BUF1.0->CROSSOVER1.0->BUF1.1->SSP5.OUTBUF2.0->CROSSOVER1.0->BUF2.0

Reply:
The first half looks good. SSP5.OUT is the end of the first pipeline and BUF2.0 is the beginning of the next pipeline (with CROSSOVER1.0 feeding BUF2.0). But why is it feeding into a crossover? Also why is it feeding back into the same buffer (BUF2.0)

What is weird is that